polkadot/runtime/parachains/src/coretime/mod.rs

```diff
@@ -245,7 +245,9 @@ impl<T: Config> OnNewSession<BlockNumberFor<T>> for Pallet<T> {
 fn mk_coretime_call(call: crate::coretime::CoretimeCalls) -> Instruction<()> {
 	Instruction::Transact {
 		origin_kind: OriginKind::Native,
-		require_weight_at_most: Weight::from_parts(1000000000, 200000),
+		// Largest call is set_lease with 1526 byte:
+		// Longest call is reserve() with 31_000_000
+		require_weight_at_most: Weight::from_parts(100_000_000, 20_000),
 		call: BrokerRuntimePallets::Broker(call).encode().into(),
```

Comment:
You have the Broker pallet in this context. Although it may have different weights than the pallet on the Relay Chain, it's IMO safer to assume some relative factor of safety than to hardcode a value.

For example how I did it with Identity Migrator with a FOS of 2, as in assume the remote weight won't be more than double for the same call.
https://github.com/paritytech/polkadot-sdk/blob/joe-people-chain/polkadot/runtime/rococo/src/impls.rs#L148

Reply:
I am actually not sure this is any safer. That file has been generated in [September](https://github.com/paritytech/polkadot-sdk/blob/69434d9a32af6103d7cbde9798cf594027fb620e/substrate/frame/broker/src/weights.rs#L21) and was never updated since then. 

I guess the safest approach is to go with a rather huge safety margin and just split it up into multiple messages.

Reply:
Yeah I know that right now one is using default weights, but a (hopefully safe) assumption is that each chain would have weights generated at roughly the same time when making a release.

Reply:
https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Frococo-rpc.polkadot.io#/extrinsics/decode/0xff00630003000100b50f03d42f000006020208af2f419c48320104ffffffffffffffffffff02e803000006020208af2f419c48320104ffffffffffffffffffff02ea03000006020208af2f419c48320104ffffffffffffffffffff02eb03000006020208af2f419c48320104ffffffffffffffffffff02ed03000006020208af2f419c48320104ffffffffffffffffffff02f503000006020208af2f419c48320104ffffffffffffffffffff02f703000006020208af2f419c48320104ffffffffffffffffffff022c07000006020208af2f419c38320104ffffffffffffffffffff0106020208af2f419c283203d0070000a724010006020208af2f419c283203d40700004f7d000006020208af2f419c283203d60700002b74000006020208af2f419c283203d7070000db87000006020208af2f419c283203db0700008f2e000006020208af2f419c283203dd070000cba6000006020208af2f419c283203df070000076b000006020208af2f419c283203ea0700009f69000006020208af2f419c283203ee070000334e000006020208af2f419c283203ef0700001779000006020208af2f419c283203f20700002fe8000006020208af2f419c283203f8070000cf93000006020208af2f419c283203fb070000cb79000006020208af2f419c2832030808000057de000006020208af2f419c2832030a080000f323010006020208af2f419c2832030b080000634b000006020208af2f419c283203270800003f42000006020208af2f419c2832032a080000a724010006020208af2f419c2832032d0800009f69000006020208af2f419c2832032f0800000f18000006020208af2f419c28320334080000cf66000006020208af2f419c2832033508000053f1000006020208af2f419c28320338080000cf93000006020208af2f419c28320339080000a724010006020208af2f419c2832033a080000a724010006020208af2f419c2832033e0800000f72000006020208af2f419c28320342080000df74000006020208af2f419c2832034708000067ec000006020208af2f419c28320349080000432f000006020208af2f419c2832034c080000d79a000006020208af2f419c2832034d0800001f53000006020208af2f419c283203bf0800007f7a000006020208af2f419c283203c80b00000798000006020208af2f419c283203050d000067ec000006020208af2f419c283203100d0000c3f9000006020208af2f419c283203110d000003ab000006020208af2f419c283203580d0000c78c000006020208af2f419c283203cc0f0000c3f9000006020208af2f419c283203bd10000003ab000006020208af2f419c283203c41000000fcc000006020208af2f419c283203c81000006fc6000006020208af2f419c283203e3100000ffea000006020208af2f419c2832030f2700002b47000006020208af2f419c1032133300

This is the XCM message that worked on coretime-rococo